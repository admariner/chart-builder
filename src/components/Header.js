// @flow
import React, { Component, Fragment } from 'react'
import { Navbar } from 'react-bootstrap'
import { observer, inject } from 'mobx-react'
import logo from './logo.svg'

import type { StoreType } from '../util/Store'

class Header extends Component<{
  store: StoreType
}> {
  render() {
    const { store } = this.props
    const { agentid, datasetid } = store

    const hasContext = !!(agentid && datasetid)

    const href = hasContext
      ? `https://data.world/${agentid}/${datasetid}`
      : 'https://data.world'

    const headerText = hasContext ? (
      <Fragment>
        <span style={{ color: '#a3afbf' }}>{agentid}/{datasetid}</span>/Chart
        Builder
      </Fragment>
    ) : (
      'Chart Builder'
    )

    return (
      <Navbar inverse staticTop fluid style={{ border: 0 }}>
        <Navbar.Header>
          <Navbar.Brand>
            <img
              alt="data.world logo"
              src={logo}
              style={{
                height: 40,
                marginTop: 9,
                width: 40,
                marginLeft: '-0.5rem'
              }}
            />
          </Navbar.Brand>
          <Navbar.Text>{headerText}</Navbar.Text>
        </Navbar.Header>
        <Navbar.Text pullRight>
          <Navbar.Link href={href}>Back to data.world</Navbar.Link>
        </Navbar.Text>
      </Navbar>
    )
  }
}

export default inject('store')(observer(Header))
