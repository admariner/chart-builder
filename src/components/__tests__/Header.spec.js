import React from 'react'
import Store from '../../util/Store'
import Header from '../Header'

it('renders', () => {
  const store = Store.create({
    location: {
      search: ''
    }
  })

  snap(<Header store={store} />)
})

it('renders with agentid/datasetid', () => {
  const store = Store.create({
    location: {
      search: 'agentid=foo&datasetid=bar'
    }
  })

  snap(<Header store={store} />)
})
