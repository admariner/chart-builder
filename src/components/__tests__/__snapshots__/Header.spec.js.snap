// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<nav
  className="navbar navbar-inverse navbar-static-top"
  style={
    Object {
      "border": 0,
    }
  }
>
  <div
    className="container-fluid"
  >
    <div
      className="navbar-header"
    >
      <img
        alt="data.world logo"
        className="navbar-brand"
        src="logo.svg"
        style={
          Object {
            "height": 40,
            "marginLeft": "-0.5rem",
            "marginTop": 9,
            "width": 40,
          }
        }
      />
      <p
        className="navbar-text"
      >
        Chart Builder
      </p>
    </div>
    <p
      className="navbar-text navbar-right"
    >
      <a
        className="navbar-link"
        href="https://data.world"
      >
        Back to data.world
      </a>
    </p>
  </div>
</nav>
`;

exports[`renders with agentid/datasetid 1`] = `
<nav
  className="navbar navbar-inverse navbar-static-top"
  style={
    Object {
      "border": 0,
    }
  }
>
  <div
    className="container-fluid"
  >
    <div
      className="navbar-header"
    >
      <img
        alt="data.world logo"
        className="navbar-brand"
        src="logo.svg"
        style={
          Object {
            "height": 40,
            "marginLeft": "-0.5rem",
            "marginTop": 9,
            "width": 40,
          }
        }
      />
      <p
        className="navbar-text"
      >
        <span
          style={
            Object {
              "color": "#a3afbf",
            }
          }
        >
          foo
          /
          bar
        </span>
        /Chart Builder
      </p>
    </div>
    <p
      className="navbar-text navbar-right"
    >
      <a
        className="navbar-link"
        href="https://data.world/foo/bar"
      >
        Back to data.world
      </a>
    </p>
  </div>
</nav>
`;
